sophia/Cargo.toml

```diff
@@ -19,6 +19,7 @@ language-tag = "0.9.0"
 lazy_static = "1.2.0"
 pest = "2.1.0"
 pest_derive = "2.1.0"
+quick-xml = "0.14.0"
```

Comment:
Do you have already tested the [YaSerde](https://github.com/media-io/yaserde/) library ?
It deserialize/serialize struct/enum into/from xml. As Serde but dedicated to XML with specific parameters to mention attributes, namespaces.
But in this case I think it will be complex to use as the model need to be generic (not limited to a struct/enum description).

Reply:
The issue of `YaSerde` (as well as any XML DOM library) here is that it would not allow streaming triples, whereas my implementation does (you don't have to wait for the whole document to start producing triples).